Addresses #635 implemented statistic infobox similar to loklak

nikhilrayaprolu opened the pull request:
Fixes issue #635 

Implemented Statistic infobox that shows top providers, authors etc



Demo Link: http://susper-pr-704.herokuapp.com/

Screenshots for the change: 

![screenshot from 2017-08-03 01-13-07](https://user-images.githubusercontent.com/15216503/28893011-90570902-77ed-11e7-94eb-7d3442c2c17d.png)
![screenshot from 2017-08-03 01-12-55](https://user-images.githubusercontent.com/15216503/28893012-905f03a0-77ed-11e7-9691-e061cf0a17e9.png)

harshit98 commented on src/app/statsbox/statsbox.component.html:
```diff
@@ -0,0 +1,15 @@
+
+      </ul>
+    </div>
+
```
Remove extra line.

harshit98 commented on src/app/statsbox/statsbox.component.html:
```diff
@@ -0,0 +1,15 @@
+            {{element.name}} <span class="badge">{{element.count}}</span>
+          </a>
+        </li>
+
```
Remove extra line.

harshit98 commented on src/app/statsbox/statsbox.component.spec.ts:
```diff
@@ -0,0 +1,39 @@
+        StatsboxComponent,
+      ],
+      providers: [
+
```
Since there are no providers, we can remove it.

harshit98 commented on src/app/statsbox/statsbox.component.ts:
```diff
@@ -0,0 +1,38 @@
+  navigation$: Observable<any>;
+  selectedelements: Array<any> = [];
+  changeurl(modifier, element) {
+
```
Remove extra line.

harshit98 commented:
@nikhilrayaprolu Please go through the suggested changes which need to be done. Since we're in the final phase of GSoC, it would be great if we maintain code quality too. :)

Marauderer97 commented on src/app/statsbox/statsbox.component.ts:
```diff
@@ -0,0 +1,38 @@
+
+  }
+  ngOnInit() {
+
```
Please remove extra line

Marauderer97 commented on src/app/statsbox/statsbox.component.ts:
```diff
@@ -0,0 +1,38 @@
+      this.querylook = Object.assign({}, query);
+      this.navigation$ = store.select(fromRoot.getNavigation);
+    });
+
```
Please remove extra line

nikhilrayaprolu commented:
@harshit98 @Marauderer97 made the requested changes.

mariobehling commented:
Demo link not working on Chrome. Page is https.
![screenshot from 2017-08-03 09-49-12](https://user-images.githubusercontent.com/1583873/28911507-132d3e4e-7831-11e7-95e7-3dc7584430ca.png)

I can see already the following:
* Use same colors as exisintg
* Use same box style and indents

Still need to test the links.

nikhilrayaprolu commented:
@mariobehling please use this new deployment link at http://susper-pr-704.herokuapp.com/

nikhilrayaprolu commented:
![image](https://user-images.githubusercontent.com/15216503/28936964-ee33f18e-78a6-11e7-94ed-23f5733bba43.png)

nikhilrayaprolu commented:
@mariobehling done with the requested changes removed Protocol and collections since they don't make a meaning for user. I will make the distribution in a new PR.

mariobehling commented:
@nikhilrayaprolu 

* What is the heading of the box? There is none currently?
* Please position the sidebar boxes more smoothly in the design. The top box is not in the same line as the top text and the second box has a big space. Reduce the space.
* Where did you get the idea for the gray boxes? They are very dominant. Please propose a less dominant design, that does not attract the attention of the user right away. The main focus of the page should always be the search results.
* You show the top ten results. Please provide an option to view more results.
* The first line below "top authors" does not show a name. What is this number related to? Undefined?

mariobehling commented:
Adding this box now also adds another feature to the page that increases the issues with the design of the site in different resolutions. See the screenshot below with a number of display issues on large screens.
![screenshot from 2017-08-04 09-05-38](https://user-images.githubusercontent.com/1583873/28957870-3cb2aa10-78f4-11e7-88d4-282b8adbafa1.png)

nikhilrayaprolu commented:
@mariobehling updated the PR:

- implemented more and less button that expands
- left no margin b/w both infobox and statistic box, which is more better in user experience.
- The statistic box is self explanatory with titles as Top Provider and Top Authors, so I think we don't need to have a heading to the box.
- Grey color is changed to green
- removed the undefined author.

Regarding the large screen the problem persists on susper.com and not generated in this PR we could solve that issue in https://github.com/fossasia/susper.com/issues/683 

![screenshot from 2017-08-05 02-11-27](https://user-images.githubusercontent.com/15216503/28986377-55f0df20-7984-11e7-955f-2ad350d7d707.png)
![screenshot from 2017-08-05 02-10-46](https://user-images.githubusercontent.com/15216503/28986378-55f1307e-7984-11e7-99d2-bf54ec443083.png)
![screenshot from 2017-08-05 02-10-31](https://user-images.githubusercontent.com/15216503/28986379-55f230be-7984-11e7-9131-011882657216.png)
![screenshot from 2017-08-05 01-46-20](https://user-images.githubusercontent.com/15216503/28986380-55f35cb4-7984-11e7-9cdc-9747fcc88b1e.png)
![screenshot from 2017-08-05 01-46-12](https://user-images.githubusercontent.com/15216503/28986381-55f9b046-7984-11e7-81f2-b949f6dabc26.png)

harshit98 commented:
@nikhilrayaprolu Looks good to me! :100: Good work! :smile:

nikhilrayaprolu commented:
@mariobehling @harshit98 I have updated this pull request ,implementing graphs feature too.
![screenshot from 2017-08-05 15-18-04](https://user-images.githubusercontent.com/15216503/28994745-1ad0b338-79f3-11e7-9214-6eebd97a2eac.png)

harshit98 commented on src/app/not-found/not-found.component.spec.ts:
```diff
@@ -31,7 +31,9 @@ import { DropdownComponent } from '../dropdown/dropdown.component';
+import {StatsboxComponent} from "../statsbox/statsbox.component";
 import {InfiniteScrollModule} from "ngx-infinite-scroll";
+import {ChartsModule} from "ng2-charts";
 
```
Why you're adding `ChartsModule` and `Statsbox` here?

nikhilrayaprolu replied:
we are importing all modules into NotFound component spec it has all components in it.

harshit98 commented on src/app/services/speech-synthesis.service.ts:
```diff
@@ -29,6 +29,7 @@ export class SpeechSynthesisService {
   pause(): void {
     const { speechSynthesis }: IWindow = <IWindow>window;
+    const { SpeechSynthesisUtterance }: IWindow = <IWindow>window;
 
```
Any reason for adding this line no. 32?

nikhilrayaprolu replied:
yes without this it is giving me `SpeechSynthesisUtterance` not found

harshit98 commented on src/app/statsbox/statsbox.component.html:
```diff
@@ -0,0 +1,34 @@
+            [labels]="lineChartLabels"
+            [options]="lineChartOptions"
+            [chartType]="lineChartType"
+            ></canvas>
```
Indent this line.

harshit98 commented on src/app/statsbox/statsbox.component.ts:
```diff
@@ -0,0 +1,114 @@
+  }
+
+  public chartHovered(e: any): void {
+    console.log(e);
```
Remove `console.log`

harshit98 commented on src/app/statsbox/statsbox.component.ts:
```diff
@@ -0,0 +1,114 @@
+  public lineChartLegend: boolean = true;
+  public lineChartType: string = 'line';
+  public chartClicked(e: any): void {
+    console.log(e);
```
Remove `console.log`

nikhilrayaprolu commented:
@mariobehling made the requested changes
![screenshot from 2017-08-08 01-15-21](https://user-images.githubusercontent.com/15216503/29043098-4d38e530-7bd7-11e7-8107-b7e4a65d4f64.png)
![screenshot from 2017-08-08 01-15-12](https://user-images.githubusercontent.com/15216503/29043099-4d3f1838-7bd7-11e7-9e9e-103b0e382bdb.png)
![screenshot from 2017-08-08 01-14-55](https://user-images.githubusercontent.com/15216503/29043102-4df5c68c-7bd7-11e7-987a-2cb2e739bd7d.png)

Regarding a monthly wise statistics, that could be implemented if we could so statistics of all results, but presently we are just showing the results of present page because to retrieving specific fields of a result could be done only when `&wt=json` but we can't do that when `&wt=yjson` but JSONP callback is supporting only yjson format. I think we can achieve that with a small change on the server side. @Orbiter 
when `wt=yjson`
![screenshot from 2017-08-08 01-18-32](https://user-images.githubusercontent.com/15216503/29043264-0eb3e642-7bd8-11e7-887f-d1cba655d3bf.png)
when `wt=json`
![screenshot from 2017-08-08 01-18-06](https://user-images.githubusercontent.com/15216503/29043275-1b7eeac0-7bd8-11e7-845e-b154917cd1e5.png)

harshit98 commented:
@nikhilrayaprolu Please follow up this commit: <a href="https://github.com/harshit98/susper.com/commit/02b9898261693aa598203ca2d9528897d824a8f1">here</a> and remove unnecessary import of modules from NotFoundComponent.

nikhilrayaprolu commented:
@harshit98 sure, doing that now.

nikhilrayaprolu commented:
@harshit98 updated the pull removing all extra modules

harshit98 commented:
Looks good to me now! We can merge this PR. Good work! :100:

mariobehling commented:
Please change "Protocol Frequency" to "Protocol Distribution" in a follow up PR as well.

nikhilrayaprolu commented:
:+1:
